src/OpticalFlowEstimator.cpp

```diff
@@ -0,0 +1,598 @@
+#include "iarc7_vision/OpticalFlowEstimator.hpp"
+
+// BAD HEADER
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#include <cv_bridge/cv_bridge.h>
+#pragma GCC diagnostic pop
+// END BAD HEADER
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/core.hpp>
+#include <opencv2/gpu/gpu.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <ros/ros.h>
+#include <tf2/LinearMath/Matrix3x3.h>
+#include <tf2/LinearMath/Quaternion.h>
+
+#include <ros_utils/SafeTransformWrapper.hpp>
+
+#include <geometry_msgs/PointStamped.h>
+#include <iarc7_msgs/OrientationAnglesStamped.h>
+
+namespace iarc7_vision {
+
+static void download(const cv::gpu::GpuMat& d_mat,
+                     std::vector<cv::Point2f>& vec)
+{
+    vec.resize(d_mat.cols);
+    cv::Mat mat(1, d_mat.cols, CV_32FC2, (void*)&vec[0]);
+    d_mat.download(mat);
+}
+
+static void download(const cv::gpu::GpuMat& d_mat, std::vector<uchar>& vec)
```

Comment:
Is this still used? I'm having trouble tracking this down. But if it is used, my assumption is that it is used to get grayscale and full color versions of the same image, and that seems wasteful as download seems like an expensive operation. But again, this is all based off of an assumption.

Reply:
And they should be named different things, in my opinion.

Reply:
They're used for downloading vectors (around line 450), not images (although both are represented as matrices).  This PR isn't for optimizing any operations (unless there's something blatantly inefficient and unnecessary), we'll be optimizing later.  We do need all of those vectors out of the GPU.